flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/strategy/TestingSchedulingTopology.java

```diff
@@ -78,6 +83,31 @@ public TestingSchedulingResultPartition getResultPartition(final IntermediateRes
 		return resultPartition;
 	}
 
+	@Override
+	public Iterable<SchedulingPipelinedRegion> getAllPipelinedRegions() {
+		return vertexRegions.values().stream().collect(Collectors.toSet());
+	}
+
+	@Override
+	public SchedulingPipelinedRegion getPipelinedRegionOfVertex(ExecutionVertexID vertexId) {
+		return vertexRegions.get(vertexId);
+	}
+
+	void generatePipelinedRegions() {
```

Comment:
Consider invoking this method lazily in `getAllPipelinedRegions()` and `getPipelinedRegionOfVertex()` instead of relying the client (test) to invoke it.

Reply:
Yes that would be better.
The only problem is that the regions cannot be refreshed if the topology is changed after invoking `getAllPipelinedRegions()`. So I will clear the cached regions on any topology changes.